### ISSUE TITLE
postional arguments vs. named parameters

### ISSUE THREAD

**Issue by bturkus** [amiaopensource/ffmpeg-artschool#6]
@nkrabben mentioned the following in a separate thread; let's continue to discuss here.

```
Any opinions on making all arguments named parameters, like this
./blend.sh -v1 path/to -v2 path/to -mode selection
instead of positional arguments like this
./blend.sh path/to  path/to selection
?

Personally i like having named parameters if only for the teaching value of
having people associate inputs with what the script is doing. Also, they're
easier for me to code with.
```

**Comment by nkrabben** [comment]
Since putting in that comment, I figured out how to make Powershell scripts do both positional and named arguments, so I'm good with coding whatever. This can be purely a discussion of what's easiest to support in code and how legible scripts should be for novices.

So for an example of where positional arguments can be confusing, chromakey.sh

There's a function inside that builds the filter string. It's called with any positional arguments after the first two
`{chromakeyFilter "${@:3}"}`

Then within that function, the arguments are referred to by the position that they were passed by
`local key="${1:-00FF00}"`

But in the larger scope, $1 refers to a filename, in this and other scripts.

I realize legibility is a relative assessment, especially since my Powershell code for validating arguments looks like this.
https://github.com/iamdamosuzuki/ffmpeg-artschool/blob/c25cd1467ce0945a38e08985bd09ce5281ae1b3e/powershell_scripts/chromakey.ps1#L27

And we'd have to figure out good argument names. E.g. the `v1`  I use is hot trash. `vid1` would be better, or another suggestion.


Finally, suggested solutions:
1. getting boilerplate getopts code working for all the scripts, or
2. immediately assigning positional arguments to variables with meaningful names, or
3. ignore, or
4. ...?

**Comment by nkrabben** [comment]
The other argument I would have towards naming arguments is so that workshop attendees just using the scripts will be reinforcing what exactly they're tweaking

`./chromakey.sh file1 file2 blue`
`./chromakey.sh -background file1 -foreground file2 -chromakey blue`

I think the second is more explanatory and less mysterious

**Comment by bturkus** [comment]
You're making good points, and I'd personally be open to this but tbh I'm not sure exactly how this kinda change would be enacted. Does it all happen in getopts? And would be willing to adapt say, the chromakey script, in this way and send in a pull request for us to compare/contrast? If we can all get on the same page and we have good starter code I'd be happy to do the grunt work. 

At the moment I'm more interested in getting usage or help instructions embedded in all the code, which again, does this happen with getopts? Looking at CUNY media microservices it appears that they have a usage function at the start of most scripts.

**Comment by nkrabben** [comment]
Agreed. Looks like there are many ways to do help text in bash. The ones I'm seeing most frequently and that seem to be the least hacky generally use getopts. https://stackoverflow.com/a/5476278

I think that ties back into this discussion, because to have help text, we'd have to:
* have getopts for some and positional for others like what [CUNY MM](https://github.com/mediamicroservices/mm/blob/master/makeframemd5) does
* go all in on getopts
* hack together something that allows for both positional and opts (this feels like it would be a big headache)

**Comment by nkrabben** [comment]
named v positional was addressed for bash in PR #12 

Powershell slightly diverging. All positional arguments are also named, because it's built into the `Param()` function.
